Improvements to Explore mode

RPicster opened the pull request:
This PR includes improvements and fixed for Explore mode and some general bug fixes and improvements.

- Improved Error handling
- The UI now uses the `reported` section on all API calls
- Improved Treemap Charts (automatic scaling labels, labels change bg-color depending on the brightness of the background cell
- "All Data" was added to the Single Resource Explorer
- The Search bar now also takes searches starting with `search ` and handle them correctly
- Added a "more" button to the search results that will take the user to Resource List Explorer
- Added a styled "Copy to Clipboard" button that gives more feedback when clicked
- Added a slight delay on the search box
- Showing a Leaf icon when the currently displayed Resource in the Single Resource Explorer is a leaf (no successors)
- The user can now manage Tags on Explore mode
- The user can now take basic Resoto actions in the Explore mode (Protect, Clean, etc)

azagaya commented on src/components/elements/element_treemap.gd:
```diff
@@ -295,52 +270,87 @@ func try(_start: int, _end: int, vertical: bool) -> float:
+
+func get_treemap_from_api(node_id:String):
+	clear_treemap()
+	var search_command = "id(\"" + node_id + "\") limit 1"
```
We can use static type for search command `var search_command := ...`

azagaya commented on src/components/elements/element_treemap_box.gd:
```diff
@@ -12,8 +13,9 @@ onready var scaler = $Center/C/Z
 func _ready():
-	text_label.text = label + "\n(" + str(value) + ")"
-	$Button.hint_tooltip = label + ": " + str(value)
+	var value_str = Utils.comma_sep(value)
```
We could use static typing here too.

azagaya commented on src/components/elements/styled/flat_copy_button.gd:
```diff
@@ -0,0 +1,42 @@
+const icon_copy = preload("res://assets/icons/icon_128_copy.svg")
+
+var orig_alpha:= 1.0
+onready var tween = $ClickEffectTween
```
Static type here too for the tween

azagaya commented on src/components/fulltext_search_menu/menu_full_text_search.gd:
```diff
@@ -114,3 +147,44 @@ func _on_FullTextSearch_gui_input(event):
+
+
+func text_to_search(_text:String):
+	var _new_search_command = ""
```
Static type here too

azagaya commented:
I left some comments, but are just details about some static typing, and that we should define a convention for string manipulation. 
Besides that is good to merge. Great work!

RPicster commented:
> I left some comments, but are just details about some static typing, and that we should define a convention for string manipulation. Besides that is good to merge. Great work!

I added typing where I was missing it and also cleaned up a bit.

While I was at it, I checked the project for typing and it seems like we have some homework ... at some point 😉
